ui/nuxeo-filter.js

```diff
@@ -178,12 +180,38 @@ import { FiltersBehavior } from './nuxeo-filters-behavior.js';
     }
 
     _evaluate(document, user, expression) {
+      let res = false;
+
       try {
-        const fn = new Function(['document', 'user'], `return ${expression};`);
-        return fn.apply(this, [document, user]);
+        if (Nuxeo.UI && Nuxeo.UI.config && Nuxeo.UI.config.expressions && Nuxeo.UI.config.expressions.eval === false) {
+          const js = new Interpreter(expression, (interpreter, scope) => {
+            // set scope
+            interpreter.setProperty(scope, 'this', interpreter.nativeToPseudo(FiltersBehavior));
+            Object.entries({ document, user }).forEach(([k, obj]) => {
+              const v = {};
+              // filter out private properties
+              Object.getOwnPropertyNames(obj)
+                .filter((p) => !p.startsWith('_'))
+                .forEach((p) => {
+                  v[p] = obj[p];
+                });
+              interpreter.setProperty(scope, k, interpreter.nativeToPseudo(v));
+            });
+            // XXX: 'this' in the scope of native functions is the interpreter instance
+            Object.assign(interpreter, FiltersBehavior);
+          });
+          js.run();
+          res = js.value;
+        } else {
+          const fn = new Function(['document', 'user'], `return ${expression};`);
+          res = fn.apply(this, [document, user]);
+        }
+        return res;
       } catch (err) {
-        return false;
+        // console.log(err);
```

Comment:
We should log the error or do something with it.

Reply:
we were not logging it before, I'm guessing because the expression will be evaluated multiple times